controllers/extendeddaemonset/controller.go

```diff
@@ -256,44 +284,32 @@ func (r *Reconciler) updateInstanceWithCurrentRS(logger logr.Logger, daemonset *
 					return newDaemonset, reconcile.Result{}, err
 				}
 			}
-
-			isPaused, reason := IsCanaryDeploymentPaused(daemonset.GetAnnotations())
-			if isPaused {
-				newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateCanaryPaused
-				newDaemonset.Status.Reason = reason
-			} else {
-				newDaemonset.Status.State = datadoghqv1alpha1.ExtendedDaemonSetStatusStateCanary
-			}
 		}
 	}
 
 	// Check if newDaemonset differs from existing daemonset, and update if so
 	if !apiequality.Semantic.DeepEqual(daemonset, newDaemonset) {
-		if updateDaemonsetSpec {
-			// Make and use a copy because undesired behaviors occur when making two update calls
-			newDaemonsetCopy := newDaemonset.DeepCopy()
-			logger.Info("Updating ExtendedDaemonSet")
-			if err := r.client.Update(context.TODO(), newDaemonsetCopy); err != nil {
-				logger.Error(err, "Failed to update ExtendedDaemonSet")
-				return newDaemonsetCopy, reconcile.Result{}, err
-			}
+		logger.Info("Updating ExtendedDaemonSet status")
 
-			// This ensures that the first client update respects the desired new status
-			newDaemonsetCopy.Status = newDaemonset.Status
-			logger.Info("Updating ExtendedDaemonSet status")
-			if err := r.client.Status().Update(context.TODO(), newDaemonsetCopy); err != nil {
-				logger.Error(err, "Failed to update ExtendedDaemonSet status")
-				return newDaemonsetCopy, reconcile.Result{}, err
-			}
-			return newDaemonsetCopy, reconcile.Result{}, nil
+		// Updating the status in any case.
+		// Make and use a copy because undesired behaviors occur when making two update calls
+		// The returned EDS instance has losted the ExtendedDaemonSet.spec update part.
```

Comment:
suggestion: `The EDS instance will lose its new spec as a result of the Status update`